hudi-client/hudi-client-common/src/test/java/org/apache/hudi/common/testutils/HoodieMetadataTestTable.java

```diff
@@ -86,15 +89,15 @@ public HoodieCommitMetadata doWriteOperation(String commitTime, WriteOperationTy
   public HoodieTestTable moveInflightCommitToComplete(String instantTime, HoodieCommitMetadata metadata) throws IOException {
     super.moveInflightCommitToComplete(instantTime, metadata);
     if (writer != null) {
-      writer.update(metadata, instantTime, false);
+      writer.update(metadata, HoodieListData.eager(Collections.EMPTY_LIST), instantTime);
```

Comment:
The RI code is using HoodieData APIs instead of casting the HoodieData into HoodieJavaRDD. This keeps the code common across Spark and Flink.

See HoodieBackedTableMetadataWriter::getRecordIndexUpdates()